lib/std/math/big/int.zig

```diff
@@ -315,7 +315,7 @@ pub const Mutable = struct {
             return;
         }
 
-        if (a.limbs.len == 1 and b.limbs.len == 1 and a.positive == b.positive) {
+        if (a.limbs.len == 1 and b.limbs.len == 1 and a.positive == b.positive and r.limbs.ptr != a.limbs.ptr and r.limbs.ptr != b.limbs.ptr) {
```

Comment:
I think a better fix would be to add a stack temporary as the result location for the `@addWithOverflow` and then store that in `r.limbs[0]` since that would allow the optimization even if `a` or `b` alias `r`.

Reply:
Much better idea!